### PR TITLE
breaking change: .well-known/jwt-issuer renamed to .well-known/jwt-vc-issuer

### PR DESCRIPTION
**Pull request by peppelinux**
~this PR resolves~ https://github.com/vcstuff/draft-terbu-sd-jwt-vc/issues/101

more discussion is needed

### CODE REVIEW

File: draft-terbu-sd-jwt-vc.md
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:303]:
```suggestion
# JWT VC Issuer Metadata {#vc-jwt-issuer-metadata}
```

**Comment by tlodderstedt** [comment]
might be a stupid question: but given the latest change from VC-SD-JWT to SD-JWT VC, shouldn’t the metadata location name be `jwt-vc-issuer` instead of `vc-jwt-issuer`?

**Review by tlodderstedt — CHANGES_REQUESTED** [review]
please also see my comment

File: draft-terbu-sd-jwt-vc.md
Comment (peppelinux) [reply draft-terbu-sd-jwt-vc.md]: done, thank you

**Review by OR13 — CHANGES_REQUESTED** [review]
I don't think this helps with readability

**Comment by danielfett** [comment]
I propose that we wait to resolve the discussion in #101 before merging this.

**Comment by awoie** [comment]
Marked as DO NOT MERGE since we have to wait for the discussion in #101 to resolve first.

File: draft-terbu-sd-jwt-vc.md
Comment (OR13) [inline draft-terbu-sd-jwt-vc.md:312]: why are we intentionally breaking compatibility with oidc?

File: draft-terbu-sd-jwt-vc.md
Comment (OR13) [inline draft-terbu-sd-jwt-vc.md:403]: provide a full example of the dereferenced resource.... something like:

```json
{
  "keys": [
  {
     "kid": "..."
  }
   ]
}
```

**Review by OR13 — CHANGES_REQUESTED** [review]
This seems to break compatibility with both DIDs and OIDC.

**Comment by OR13** [comment]
I added a related PR here: https://github.com/w3c/vc-jwt/pull/111

**Review by Sakurann — CHANGES_REQUESTED** [review]
i do not think we have enough data points to make this decision, yet. this mechanism `/.well-known/jwt-issuer/` is meant to be a generic mechanism, applicable beyond only VCs and I don't think it conflicts with other .well-known paths like in federation (partially because we do not have clarity around what other metadata to put under that path other thank JWK...)

File: draft-terbu-sd-jwt-vc.md
Comment (Sakurann) [reply draft-terbu-sd-jwt-vc.md:312]: OIDC uses a different path, so I don't think it breaks compatibility with it. (this path is being defined in this spec, because we were surprised to find that a generic jwt-issuer .well-known path has not been defined, but at the same time maybe it was not defined because it is too generic and this path needs to be scoped to jwt-vcs, not sure yet)

Comment (tlodderstedt) [reply draft-terbu-sd-jwt-vc.md:312]: @OR13 A SD-JWT VC is not an ID Token and a verifier does not know how a SD-JWT VC was issued. So I don't see the relationship to OIDC: Could you please explain what "breaking compatibility with OIDC" means?

**Review by selfissued — CHANGES_REQUESTED** [review]
I would leave the name `.well-known/jwt-issuer`.  This is a general-purpose capability usable by other kinds of JWTs.  It just happens to being defined in this spec.

There's nothing wrong with that.  In fact, specs often do that, to the benefit of others wanting to use these more-general-purpose facilities.  They get put in IANA registries, from which their definitions are discoverable.

**Comment by peppelinux** [comment]
@selfissued what about having jwt_issuer as a oidc federation metadata type?

If this well known resource Is a generic purpose metadata that every kind of entity can expose, how to check the protocol metadata It carries?

It might be and OpenID provider and a relying party and an RS or whatever, without a unique identifier It wouldnt be good for metadata but Just for Key attestation

So the difference between this and x509 would be just the data format.

Or, differently, in jwt-issuer there would space for multiple metadata with unique type identifiers like the federation EC?

**Comment by OR13** [comment]
Why not define, `.well-known/jwks.json`... doesn't seem like jwt or issuer are needed...

**Comment by peppelinux** [comment]
> Why not define, `.well-known/jwks.json`... doesn't seem like jwt or issuer are needed...

That's the same though I had, if It Is Just for Key attestation let's give to it the semantic It should

**Comment by peppelinux** [comment]
Ok dears, may I change the endpoint name to `.well-known/jwks.json` in this PR and ask again your kindly revision?

**Comment by selfissued** [comment]
`.well-known/jwt-issuer` is a better name than `.well-known/jwks.json`.  The latter implies that the only claim that will be in the metadata is the `jwks_uri` claim, which won't be the case for some use cases.  Just like `.well-known/openid-configuration` and `.well-known/oauth-authorization-server`, yes, we'll have `jwks_uri`, but we also want the metadata to be extensible.

The existing name is good.

**Comment by peppelinux** [comment]
name are very often semantical or cosmetical and driven by personal tastes or objective meanings, that's just a name.

I would like to ask which are the scopes of this endpoint, if it is just for key attestation or also for metadata.
I assume the second. Since it is also for metadata, I assume here, how the metadata of a formless and abstract entity could be carried in that endpoint?

my answer is, by defining several metadata types, as already done in [federation entity configuration](https://openid.net/specs/openid-connect-federation-1_0.html#name-metadata-type-identifiers) 

but this seems not clear, or out of scope for now, or let them think about this later on.
Well, without the big picture I find very difficult giving a name to an endpoint, since the name should represent its scopes and values

**Comment by tplooker** [comment]
> Why not define, `.well-known/jwks.json`... doesn't seem like jwt or issuer are needed...

I like this more, having a well known location to associate a set of JWKs to seems generally useful, if you have additional metadata to communicate to parties that should be served from higher level well-known endpoints that are defined for that protocol / application.

**Comment by peppelinux** [comment]
if the purpose of this endpoint is only for serving jwks, I think `.well-known/jwk-set` is a good choice

**Comment by TakahikoKawasaki** [comment]
Any path name would be fine (e.g., `/.well-known/jwks.json`, `/.well-known/jwk-set`, `/.well-known/jwks`, ...), but I think it's a good idea to assign a well-known path to the endpoint that returns the JWK Set document.

Defining such a well-known path will not prevent the definition of `/.well-known/jwt-issuer`. Both paths can coexist. The discussion for `/.well-known/jwt-issuer` can be revisisted in the future if the need arises.

**Comment by OR13** [comment]
You could remove the content type from the resource, and use the accept header.

GET `/.well-known/keys`

--accept application/jwk-set+json
--accept application/cose-key-set

**Comment by peppelinux** [comment]
> You could remove the content type from the resource identifier, and use the accept header.
> 
> GET `/.well-known/keys`
> 
> --accept application/jwk-set+json --accept application/cose-key-set

I love this, however I'd prefer two separate endpoints for two different format, since without the definition of any content-type in the request would return a default value, then: Jose or cose Key set?

**Comment by OR13** [comment]
Server reserves the right to set the content type header or throw a content ntyoe unsupported error, but you can also probably say jwk-set+json is the default when requesting the endpoint be reserved....

openid-configuration is known to be application/json, and didn't reserve

`/.well-known/openid-configuration.json`

**Comment by peppelinux** [comment]
overseeded by https://github.com/oauth-wg/oauth-sd-jwt-vc/pull/189

**Pull request closed without merging**
